pipeline/tasks/publish_tasks.py

```diff
@@ -56,3 +58,15 @@ def execute(self, repo_url, username, password, publish_env,
 
     def validate(self):
         return []
+
+
+class GitHubPushTask(task_base.TaskBase):
+    """Uploads to a GitHub repository"""
+
+    def execute(self, owner, branch, username, password, publish_env,
+                final_repo_dir, message):
+        github_utils.push_dir_github(final_repo_dir, username, password, owner,
```

Comment:
Name the first param to "dir_to_push", and use "rebind" (http://docs.openstack.org/developer/taskflow/atoms.html) to map "final_repo_dir" to "dir_to_push" when creating the github publish task. By doing this, we don't force the final_repo_dir to be the input/output name.